core/packages/contracts/contracts/ERC20App.sol

```diff
@@ -109,13 +96,16 @@ contract ERC20App is AccessControl {
             revert MinimumAmount();
         }
 
-        if (_amount > balances[_token]) {
-            revert InsufficientBalance();
-        }
-
-        balances[_token] = balances[_token] - _amount;
-        IERC20(_token).safeTransfer(_recipient, _amount);
-
+        vault.withdraw(_recipient, _token, _amount);
         emit Unlocked(_token, _sender, _recipient, _amount);
     }
+
+    function transferVaultOwnership(
```

Comment:
ditto. Don't want to touch governance/upgrade stuff until we have a sense on how XCMv3 will change things.